### PR TITLE
chore(kubernetes platform): Rework Kubernetes YAMLs to present multiple deployment modes

### PR DESCRIPTION
**Pull request by MOZGIII**
Implements the first part of the #5042, does not close it yet.

After this is complete, the https://github.com/timberio/vector-k8s-examples needs to be updated too.

### CODE REVIEW

File: distribution/kubernetes/vector-all/resources.yaml [hunk 1]
```diff
@@ -117,7 +117,7 @@ spec:
         - name: vector
           securityContext:
             {}
-          image: "timberio/vector:latest-alpine"
```
Comment (eeyun) [inline distribution/kubernetes/vector-all/resources.yaml:120]: I'm curious why the switch from alpine to debian?

File: scripts/kubernetes-yaml/vector-all/values.yaml [hunk 2]
```diff
@@ -2,8 +2,6 @@ vector-agent:
   image:
     # Use the latest release image.
     version: latest
-    # Use the alpine flavor of vector image.
-    base: alpine
```
Comment (eeyun) [inline scripts/kubernetes-yaml/vector-all/values.yaml:6]: Is this fine to just remove or should it be replaced with debian as well?

**Review by eeyun — APPROVED** [review]
Changes look fine other than a couple simple questions.

File: distribution/kubernetes/vector-all/resources.yaml (hunk 1 above)
Comment (MOZGIII) [reply distribution/kubernetes/vector-all/resources.yaml:120]: `alpine` uses `musl`, and musl has some perf issues. `debian` uses `glibc`.
`debian` shown about 20%  higher throughput than `alpine` in our load tests.

File: scripts/kubernetes-yaml/vector-all/values.yaml (hunk 2 above)
Comment (MOZGIII) [reply scripts/kubernetes-yaml/vector-all/values.yaml:6]: Just removing it is sufficient. The semanics here is we want to use whatever the default is in the Helm chart, and avoid special-casing for the geneated YAMLs

**Pull request merged**